PROVISIONING-METAL.md

```diff
@@ -0,0 +1,169 @@
+Interface names are expected to be correct as per `udevd` naming, no interface naming or matching is supported.
+
+#### Supported interface settings
+
+* `primary` (boolean): Uses the interface as the primary network interface, kubelet will use this interface's IP to register.  If none of the interfaces has `primary` set, we choose the first interface in the file.
+* `dhcp4` (boolean or map): Turns on DHCP4 for the interface.  If additional DHCP4 configuration is required, the following settings are supported and may be provided as a map with the following keys:
+  * `enabled` (boolean, required): Enables DHCP4.
+  * `route-metric` (integer): Prioritizes routes by setting values for preferred interfaces.
+  * `optional` (boolean): the system will request a lease using this protocol, but will not wait for a valid lease to consider this interface configured.
+* `dhcp6` (boolean or map): Turns on DHCP6 for the interface.  If additional DHCP6 configuration is required, the following settings are supported and may be provided as a map with the following keys:
+  * `enabled` (boolean, required): Enables DHCP6.
+  * `optional` (boolean): the system will request a lease using this protocol, but will not wait for a valid lease to consider this interface configured.
+
+Example `net.toml` with comments:
+```toml
+version = 1
+
+# "eno1" is the interface name
+[eno1]
+# Users may turn on dhcp4 and dhcp6 via boolean
+dhcp4 = true
+dhcp6 = true
+primary = true
+
+# "eno2" is the second interface in this example
+[eno2.dhcp4]
+# `enabled` is a boolean and is a required key when
+# setting up DHCP this way
+enabled = true
+# Route metric may be supplied for ipv4
+route-metric = 200
+
+[eno2.dhcp6]
+enabled = true
+optional = true
+```
+
+> **An additional note on network device names**
+> 
+> Bottlerocket’s interface name policies are [specified in this file](https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/release/80-release.link#L6); with name precedence in the following order: onboard, slot, path.
+> What this means is that udev finds and generates the names for the device, and precedence is given in that order.
+> Typically on-board devices are named `eno*`, hot-plug devices are named `ens*`, and if neither of those names are able to be generated, the “path” name is given, i.e `enp*s*f*`.
+
+### Boot Configuration
+
+Bottlerocket uses a neat feature of the Linux kernel called [Boot Configuration](https://www.kernel.org/doc/html/latest/admin-guide/bootconfig.html), which allows a user to pass additional arguments to the kernel command line at runtime.
+The most immediate use of this feature for most users is setting `console` settings so boot messages can be seen on the appropriate consoles.
+
+In order to make use of this feature, one must create an initrd with the desired settings encoded inside it.
+(Bottlerocket technically doesn't use an initrd, so the file is empty save for the encoded boot config data)
```

Comment:
This is confusing - we say to create an initrd and then say that Bottlerocket doesn't need it.